component/component.go

```diff
@@ -40,6 +51,18 @@ type Host interface {
 	// Context returns a context provided by the host to be used on the component
 	// operations.
 	Context() context.Context
+
+	// GetFactory of the specified kind. Returns the factory for a component type.
+	// This allows components to create other components. For example:
+	//   func (r MyReceiver) Start(host component.Host) error {
+	//     apacheFactory := host.GetFactory(KindReceiver,"apache").(component.ReceiverFactory)
+	//     receiver, err := apacheFactory.CreateMetricsReceiver(...)
+	//     ...
+	//   }
+	// GetFactory can be called by the component anytime after Start() begins and
+	// until Shutdown() is called. Note that the component is responsible for destroying
+	// other components that it creates.
+	GetFactory(kind Kind, componentType string) Factory
```

Comment:
Do you think we should have a `type ComponentType string` to make sure we can extend that in the future if needed? Maybe just an issue now.

Reply:
I will look into this separately since it is declared a string in many other places and changing it may break other code that imports us.
Issue added https://github.com/open-telemetry/opentelemetry-collector/issues/762